tests/e2e/device/e2e_device_commands.c

```diff
@@ -780,6 +785,150 @@ static uint32_t prvE2ETestVerifyDesiredPropertiesCommandExecute( E2E_TEST_COMMAN
+                                                                &xJSONWriter ) != eAzureIoTSuccess ) ||
+        ( AzureIoTJSONWriter_AppendEndObject( &xJSONWriter ) != eAzureIoTSuccess ) )
+    {
+        RETURN_IF_FAILED( e2etestE2E_TEST_FAILED, "Failed to build response with status" );
+    }
+    else if( AzureIoTHubClient_SendPropertiesReported( pxAzureIoTHubClient,
+                                                       ucScratchBuffer2,
+                                                       ( uint32_t ) AzureIoTJSONWriter_GetBytesUsed( &xJSONWriter ),
+                                                       &ulRequestId ) != eAzureIoTSuccess )
+    {
+        RETURN_IF_FAILED( e2etestE2E_TEST_FAILED, "Failed to send reported properties" );
+    }
+    else if( AzureIoTHubClient_ProcessLoop( pxAzureIoTHubClient,
+                                            e2etestPROCESS_LOOP_WAIT_TIMEOUT_IN_SEC ) != eAzureIoTSuccess )
+    {
+        RETURN_IF_FAILED( e2etestE2E_TEST_FAILED, "recevice timeout" );
+    }
+    else if( AzureIoTHubClient_SendTelemetry( pxAzureIoTHubClient,
+                                              xCMD->pulReceivedData,
+                                              xCMD->ulReceivedDataLength,
+                                              NULL, eAzureIoTHubMessageQoS1, &usTelemetryPacketID ) != eAzureIoTSuccess )
+    {
+        RETURN_IF_FAILED( e2etestE2E_TEST_FAILED, "Failed to send response" );
+    }
+    else if( ( prvWaitForPuback( pxAzureIoTHubClient, usTelemetryPacketID ) ) )
+    {
+        RETURN_IF_FAILED( e2etestE2E_TEST_FAILED, "Telemetry message PUBACK never received!" );
+    }
+
+    return e2etestE2E_TEST_SUCCESS;
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * Execute Report component properties command
+ *
+ * e.g of command issued from service:
+ *   {"method":"report_component_properties","component":"test","desired_property_key":"temp","desired_property_value":"7664262703"}
+ *
+ **/
+static uint32_t prvE2ETestReportComponentPropertiesCommandExecute( E2E_TEST_COMMAND_HANDLE xCMD,
+                                                                   AzureIoTHubClient_t * pxAzureIoTHubClient )
+{
+    uint32_t ulStatus;
+    uint16_t usTelemetryPacketID;
+    uint32_t ulRequestId;
+    az_span xJsonSpan = az_span_create( ( uint8_t * ) xCMD->pulReceivedData, xCMD->ulReceivedDataLength );
+    az_span xKey = az_span_create( ucScratchBuffer, e2etestMESSAGE_BUFFER_SIZE / 3 );
+    az_span xValue = az_span_create( ucScratchBuffer + e2etestMESSAGE_BUFFER_SIZE / 3, e2etestMESSAGE_BUFFER_SIZE / 3 );
+    az_span xComponent = az_span_create( ucScratchBuffer + 2 * e2etestMESSAGE_BUFFER_SIZE / 3, e2etestMESSAGE_BUFFER_SIZE / 3 );
+    AzureIoTJSONWriter_t xJSONWriter;
+
+    ulStatus = prvGetValueForKey( xJsonSpan, e2etestCOMPONENT_KEY, &xComponent );
+    RETURN_IF_FAILED( ulStatus, "failed to find component!" );

```

Comment:
Same here can just do a find all and replace
```suggestion
    RETURN_IF_FAILED( ulStatus, "Failed to find component!" );
```